### PR TITLE
[ie/niconico] Fix NiconicoUserIE

### PR DESCRIPTION
**Pull request by Wesley107772**
### Description of your *pull request* and other information

Updated the API version for NiconicoUserIE, as v1 seemingly stopped working some time back. Also improved the _VALID_URL string to allow for the /video pathname

Partially addresses #10084

NiconicoUserIE, as it gave a 404 error before; as well as its _VALID_URL, as links with /video at the end would default to the generic extractor.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/niconico.py [hunk 1]
```diff
@@ -869,15 +869,15 @@ def _real_extract(self, url):
 
 
 class NiconicoUserIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?nicovideo\.jp/user/(?P<id>\d+)/?(?:$|[#?])'
+    _VALID_URL = r'https?://(?:www\.)?nicovideo\.jp/user/(?P<id>\d+)(?:(/video)|/?)/?(?:$|[#?])'
```
Comment (grqz) [inline yt_dlp/extractor/niconico.py:872]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?nicovideo\.jp/user/(?P<id>\d+)(?:/video)?/?(?:$|[#?])'
```

**Review by bashonly — APPROVED** [review]

**Comment by Wesley107772** [comment]
Is this ready to be merged in its current state or is there anything else I need to do (e.g. make issues to link to this PR)?

**Comment by bashonly** [comment]
> Is this ready to be merged in its current state or is there anything else I need to do (e.g. make issues to link to this PR)?

Are there any issues that this closes? Please do if so

The test was failing; please make sure to always run the extractor download test(s) if you are fixing an extractor. The failure was due to an old bug unrelated to your changes so I went ahead and fixed it

**Comment by Wesley107772** [comment]
> > Is this ready to be merged in its current state or is there anything else I need to do (e.g. make issues to link to this PR)?
> 
> Are there any issues that this closes? Please do if so
> 
> The test was failing; please make sure to always run the extractor download test(s) if you are fixing an extractor. The failure was due to an old bug unrelated to your changes so I went ahead and fixed it

I don't believe there are any open issues related to these changes, I was just wondering. This is my first pull request so I wasn't sure whether it skipping those two tests was intentional or not. Thanks for the fix

**Comment by bashonly** [comment]
>  This is my first pull request so I wasn't sure whether it skipping those two tests was intentional or not.

There is only one test for `NicoNicoUser` and it wasn't being skipped (which is OK), it was failing (which is bad). No worries though. Another note for the future: please don't force-push to your PR branch. You can make as many commmits as you want, they will all be squashed into a single new clean commit upon merge. (Again, no worries: this is something that's not in the contributing guide and needs to be added asap)

Thanks for the PR, nice work!

**Pull request merged**